diff -Naur hacks/pacman_ai.c.orig hacks/pacman_ai.c
--- hacks/pacman_ai.c.orig
+++ hacks/pacman_ai.c
@@ -703,6 +703,37 @@
     p->lastbox = dir;
 }
 
+static void
+pac_user (pacmangamestruct * pp, pacmanstruct * p)
+{
+    int dir = pp->user_dir;
+
+    /*
+        pp->user_dir_changed = 1;
+    */
+
+    if ( pacman_check_pos (pp, p->row + dirvecs[dir].dy,
+                        p->col + dirvecs[dir].dx, 0) )
+    {
+       
+        p->nextrow = p->row + dirvecs[dir].dy;
+        p->nextcol = p->col + dirvecs[dir].dx;
+        p->lastbox = dir;
+
+    } else {
+        dir = pp->user_dir_last;
+        if ( pacman_check_pos (pp, p->row + dirvecs[dir].dy,
+                            p->col + dirvecs[dir].dx, 0) )
+        {
+           
+            p->nextrow = p->row + dirvecs[dir].dy;
+            p->nextcol = p->col + dirvecs[dir].dx;
+            p->lastbox = dir;
+
+        }
+    }
+}
+
 static int
 pac_get_vector_screen (pacmangamestruct * pp, pacmanstruct * p,
                        const int x, const int y, int *vx, int *vy)
@@ -851,6 +882,7 @@
         p->justate = 0;
     }
 
+#if 0
     if (!(pacman_trackmouse && pac_trackmouse (mi, pp, p))) {
         /* update pacman */
         switch (p->aistate) {
@@ -870,6 +902,8 @@
             break; /* Don't move */
         }
     }
+#endif
+    pac_user (pp, p);
 
     p->cfactor = GETFACTOR (p->nextcol, p->col);
     p->rfactor = GETFACTOR (p->nextrow, p->row);
diff -Naur hacks/pacman.c.orig hacks/pacman.c
--- hacks/pacman.c.orig
+++ hacks/pacman.c
@@ -55,7 +55,6 @@
 
 #	define UNIFORM_COLORS
 #	define BRIGHT_COLORS
-#   define pacman_handle_event 0
 #	include "xlockmore.h"   /* in xscreensaver distribution */
 #   include <assert.h>
 #else /* STANDALONE */
@@ -737,6 +736,54 @@
   XFillRectangle(d,w,g,xl,yl,xs,ys)
 
 
+#include <X11/keysymdef.h>
+
+ENTRYPOINT Bool
+pacman_handle_event (ModeInfo * mi, XEvent *ev)
+{
+    pacmangamestruct *pp = &pacman_games[MI_SCREEN (mi)];
+    if (ev->xany.type == KeyPress) {
+        int index = 0;
+        KeySym keysym = XKeycodeToKeysym(MI_DISPLAY(mi), ev->xkey.keycode, index);
+        /*switch (ev->xkey.keycode) {*/
+        switch (keysym) {
+            case XK_Left:
+                if (pp->user_dir != 0) {
+                    pp->user_dir_changed = 1;
+                    pp->user_dir_last = pp->user_dir;
+                    pp->user_dir = 0;
+                }
+                else pp->user_dir_changed = 0;
+                return True;
+            case XK_Right:
+                if (pp->user_dir != 2) {
+                    pp->user_dir_changed = 1;
+                    pp->user_dir_last = pp->user_dir;
+                    pp->user_dir = 2;
+                }
+                else pp->user_dir_changed = 0;
+                return True;
+            case XK_Up:
+                if (pp->user_dir != 3) {
+                    pp->user_dir_changed = 1;
+                    pp->user_dir_last = pp->user_dir;
+                    pp->user_dir = 3;
+                }
+                else pp->user_dir_changed = 0;
+                return True;
+            case XK_Down:
+                if (pp->user_dir != 1) {
+                    pp->user_dir_changed = 1;
+                    pp->user_dir_last = pp->user_dir;
+                    pp->user_dir = 1;
+                }
+                else pp->user_dir_changed = 0;
+                return True;
+        }
+    }
+    return False;
+}
+
 /* Draws the pacman sprite, removing the previous location. */
 #if defined(USE_PIXMAP)
 
diff -Naur hacks/pacman.h.orig hacks/pacman.h
--- hacks/pacman.h.orig
+++ hacks/pacman.h
@@ -221,6 +221,10 @@
     bonus_dot bonus_dots[NUM_BONUS_DOTS];
     struct tiles *tiles;
 
+    int user_dir;
+    int user_dir_changed;
+    int user_dir_last;
+
 } pacmangamestruct;
 
 extern pacmangamestruct *pacman_games;
